cpp/src/arrow/io/file.cc

```diff
@@ -510,16 +519,25 @@ class MemoryMappedFile::MemoryMap : public MutableBuffer {
 
  private:
   // Initialize the mmap and set size, capacity and the data pointers
-  Status InitMMap(int64_t initial_size, bool resize_file = false) {
+  Status InitMMap(int64_t initial_size, bool resize_file = false,
+                  const int64_t length = 0, const int64_t offset = 0) {
     if (resize_file) {
       RETURN_NOT_OK(internal::FileTruncate(file_->fd(), initial_size));
     }
     DCHECK(data_ == nullptr && mutable_data_ == nullptr);
-    void* result = mmap(nullptr, static_cast<size_t>(initial_size), prot_flags_,
-                        map_mode_, file_->fd(), 0);
+
+    size_t mmap_length = static_cast<size_t>(initial_size);
+    if (length > 0 && length < initial_size) {
```

Comment:
If `length` has an invalid value (e.g. greater than the file size), we should return an error instead of silently creating a smaller map, IMHO.

Reply:
done